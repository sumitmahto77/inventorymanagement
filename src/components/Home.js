import React from 'react'
import Products from './Products';
import Footer from './Footer';
import MenuItems from './MenuItems';

const Home = () => {
  
  return (
    <div>
      <MenuItems/>
      <Products/>
      <Footer/>       
    </div>
  );
}

export default Home