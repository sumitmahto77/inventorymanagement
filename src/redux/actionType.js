// // export const GET_USERS ="GET_USERS";
// // export const ADD_USER = "ADD_USER";

// export const LOGIN_START = "LOGIN_START";
// export const LOGIN_SUCCESS = "LOGIN_SUCCESS";
// export const LOGIN_FAIL = "LOGIN_FAIL";

// export const LOGOUT_USER = "LOGOUT_USER";

// export const REGISTER_START = "REGISTER_START";
// export const REGISTER_SUCCESS = "REGISTER_SUCCESS";
// export const REGISTER_FAIL = "LOGIN_FAIL";



