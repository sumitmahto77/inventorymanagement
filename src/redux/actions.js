import * as types from './actionType';
import axios from 'axios';

// const getUsers = (users) => ({
//     type: types.GET_USERS,
//     payload: users,
// });

// const userAdded = () =>({
//     type:types.ADD_USER,

// });

// export const loadUsers = () => {
//     return function(dispatch){
//         axios.get("http://localhost:4000/users")
//         .then((response) => {
//             console.log("response", response);
//             dispatch(getUsers(response.data));
//         })
//         .catch((error) => console.log(error));
//     };
// };

// export const addUser = (user) => {
//     return function (dispatch){
//         axios
//             .post("http://localhost:4000/users",user)
//             .then((response) =>{
//                 console.log("response",response);
//                 dispatch(userAdded());
//             })
//             .catch((error) => console.log(error));
//     };
// };


const loginStart = () => ({
    type : types.LOGIN_START,
});

const loginSuccess = (token) => ({
    type : types.LOGIN_SUCCESS,
    payload : token,
});

const loginFail = (error) => ({
    type : types.LOGIN_FAIL,
    payload : error,
});

const registerStart = () => ({
    type : types.REGISTER_START,
});

const registerSuccess = (token) => ({
    type : types.REGISTER_SUCCESS,
    payload : token,
});

const registerFail = (error) => ({
    type : types.REGISTER_FAIL,
    payload : error,
});

export const logout = () =>({
    type : types.LOGOUT_USER,

})


export const loginInitiate = (email,password) => {
    return function(dispatch){
        dispatch(loginStart());
        axios
        .post("http://localhost:4000/login",{
            email,
            password,
        })
        .then((response) => {
            console.log("response", response);
            dispatch(loginSuccess(response.data.access_token))
            
        })
        .catch((error) => dispatch(loginFail(error.response.data.message)));
    };
};
export const registerInitiate = (user) => {
    return function(dispatch){
        dispatch(registerStart());
        axios
        .post("http://localhost:4000/users",user)
        .then((response) => {
            console.log("response", response);
            dispatch(registerSuccess(response.data))
            
        })
        .catch((error) => dispatch(registerFail(error.response.data.message)));
    };
};
