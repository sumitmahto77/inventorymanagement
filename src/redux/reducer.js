import * as types from './actionType';

const initialAuthState ={
    user : null,
    loading : false,
    error : null,
    token : null
};

const initialProdState = {
    products : [],
    loading : true,
};

export const authReducer = (state = initialAuthState, action) => {
    switch (action.type){
        case types.LOGIN_START:
        case types.REGISTER_START:
            return{
                ...state,
                loading : true,
            };
        case types.LOGIN_SUCCESS:
        case types.REGISTER_SUCCESS:
            return{
                ...state,
                loading: false,
                user :action.payload.user,
                token : action.payload.accessToken,
            };
        case types.LOGIN_FAIL:
        case types.REGISTER_FAIL:
            return{
                ...state,
                loading :false,
                error : action.payload,
            };

        case types.LOGOUT_USER:
            return{
                ...state,
                user : null,
                token :null,
            };

        default:
            return state;
    }
};

export const prodReducer =(state=initialProdState,action) => {
    switch (action.type){
        case types.GET_PRODUCTS:
            return {
                ...state,
                products : action.payload,
                loading: false,
            };
        case types.ADD_PRODUCT:
            return{
                ...state
            }
        case types.DELETE_PRODUCT:
            return{
                ...state,
                loading :false,
            }

        default:
            return state;
    }

}
