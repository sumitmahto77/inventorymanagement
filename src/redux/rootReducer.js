import {combineReducers} from "redux";
import {persistReducer} from "redux-persist";
import storage from "redux-persist/lib/storage";
import {authReducer,prodReducer} from "./reducer";




// const rootReducer = combineReducers({
//     users: usersReducers
// });

// export default rootReducer; 



const persistConfig = {
    key : "root",
    storage,
    whitelist: ["auth","prod"]
}


const rootReducer = combineReducers({
    auth : authReducer,
    prod : prodReducer,
});

export default persistReducer(persistConfig, rootReducer); 

