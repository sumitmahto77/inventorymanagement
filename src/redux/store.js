import {createStore, applyMiddleware} from "redux";
import {composeWithDevTools} from 'redux-devtools-extension';
import logger from 'redux-logger';
import reduxThunk from 'redux-thunk';
import {persistStore} from "redux-persist";
import rootReducer from './rootReducer';

const middlewares =[reduxThunk];

if (process.env.NODE_ENV === "development"){
    middlewares.push(logger)
}

export const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(...middlewares)) );

export const persistor = persistStore(store); 


