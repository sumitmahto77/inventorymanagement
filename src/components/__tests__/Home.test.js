import renderer from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';
import {Provider} from 'react-redux';
import {store} from '../../redux/store';
import Home from '../Home';

describe('Home', () =>{
    test('Renders correctly', () => {
        const tree = renderer.create(
        <Provider store={store}>
        <Router>
            <Home/>
        </Router>
        </Provider>
        );
        expect(tree).toMatchSnapshot();
    });
});