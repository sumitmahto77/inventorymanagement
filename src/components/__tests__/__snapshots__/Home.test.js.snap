// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Renders correctly 1`] = `
<div>
  <div
    className="sc-kgflAQ iDMffT"
  >
    <a
      href="/addproduct"
      onClick={[Function]}
    >
      <button
        className="sc-fLlhyt hSNtAZ"
      >
        add Product
      </button>
    </a>
    <a
      href="/delete"
      onClick={[Function]}
    >
      <button
        className="sc-fLlhyt hSNtAZ"
      >
        delete product
      </button>
    </a>
    <a
      href="/chart"
      onClick={[Function]}
    >
      <button
        className="sc-fLlhyt hSNtAZ"
      >
        most viewed products
      </button>
    </a>
  </div>
  <div
    className="sc-iBkjds kxbXLa"
  >
    <div
      className="sc-ftvSup bbamNW"
    >
      <input
        className="sc-papXJ kGOtXi"
        onChange={[Function]}
        placeholder="Search for..."
        type="search"
        value=""
      />
      <button
        className="sc-jqUVSM cNaTdX"
        onClick={[Function]}
      >
        Customize
      </button>
    </div>
    <div
      className="sc-ftvSup bbamNW"
    />
  </div>
  <div
    className="sc-crXcEl cIzLcc"
  >
    <div
      className="sc-evZas cEgShN"
    >
      <h1
        className="sc-breuTD"
      >
        IMS
      </h1>
      <p
        className="sc-ksZaOG rbNyn"
      >
        IMS gives you an all round solution for your inventory management problems.
      </p>
      <div
        className="sc-hAZoDl ihvjrF"
      >
        <div
          className="sc-fnykZs jRlJkq"
          color="3B5999"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="FacebookIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
            />
          </svg>
        </div>
        <div
          className="sc-fnykZs beoJoR"
          color="E4405F"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="InstagramIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
            />
          </svg>
        </div>
        <div
          className="sc-fnykZs hoUebJ"
          color="55ACEE"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="TwitterIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
            />
          </svg>
        </div>
        <div
          className="sc-fnykZs klShix"
          color="E60023"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="PinterestIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M9.04 21.54c.96.29 1.93.46 2.96.46a10 10 0 0 0 10-10A10 10 0 0 0 12 2 10 10 0 0 0 2 12c0 4.25 2.67 7.9 6.44 9.34-.09-.78-.18-2.07 0-2.96l1.15-4.94s-.29-.58-.29-1.5c0-1.38.86-2.41 1.84-2.41.86 0 1.26.63 1.26 1.44 0 .86-.57 2.09-.86 3.27-.17.98.52 1.84 1.52 1.84 1.78 0 3.16-1.9 3.16-4.58 0-2.4-1.72-4.04-4.19-4.04-2.82 0-4.48 2.1-4.48 4.31 0 .86.28 1.73.74 2.3.09.06.09.14.06.29l-.29 1.09c0 .17-.11.23-.28.11-1.28-.56-2.02-2.38-2.02-3.85 0-3.16 2.24-6.03 6.56-6.03 3.44 0 6.12 2.47 6.12 5.75 0 3.44-2.13 6.2-5.18 6.2-.97 0-1.92-.52-2.26-1.13l-.67 2.37c-.23.86-.86 2.01-1.29 2.7v-.03z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="sc-fEOsli iyaMWH"
    >
      <h3
        className="sc-bjUoiL dvqGvw"
      >
        Useful Links
      </h3>
      <ul
        className="sc-idiyUo drCieh"
      >
        <li
          className="sc-dIouRR bQuzZj"
        >
          Home
        </li>
        <li
          className="sc-dIouRR bQuzZj"
        >
          My Account
        </li>
      </ul>
    </div>
    <div
      className="sc-hHLeRK IhsMK"
    >
      <h3
        className="sc-bjUoiL dvqGvw"
      >
        Contact
      </h3>
      <div
        className="sc-dmRaPn gTUlTu"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="RoomIcon"
          focusable="false"
          style={
            Object {
              "marginRight": "10px",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
          />
        </svg>
         Sarjapur , Bangalore
      </div>
      <div
        className="sc-dmRaPn gTUlTu"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="PhoneIcon"
          focusable="false"
          style={
            Object {
              "marginRight": "10px",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02l-2.2 2.2z"
          />
        </svg>
         +1 234 56 78
      </div>
      <div
        className="sc-dmRaPn gTUlTu"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MailOutlineIcon"
          focusable="false"
          style={
            Object {
              "marginRight": "10px",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
          />
        </svg>
         contact@wipro.com
      </div>
    </div>
  </div>
</div>
`;
